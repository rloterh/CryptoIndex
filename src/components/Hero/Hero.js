import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../assets/logo.png';
import './Hero.scss';

const Hero = (props) => {
  const { btcDominance, ethDominance } = props;
  return (
    <div className="hero-container d-flex" data-testid="global-container">
      <div className="d-data-1">
        <img src={logo} alt="logo" className="logo" />
      </div>
      <div className="d-data-2">
        <h2>::Currency Dominance::</h2>
        <p>
          {'BTC: '}
          {btcDominance}
          %
        </p>
        <p>
          {'ETH: '}
          {ethDominance}
          %
        </p>
      </div>
    </div>
  );
};

Hero.propTypes = {
  btcDominance: PropTypes.string.isRequired,
  ethDominance: PropTypes.string.isRequired,
};

export default Hero;
