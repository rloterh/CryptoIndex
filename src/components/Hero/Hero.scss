.hero-container {
  display: flex;
  padding: 2vh;
  font-family: 'Roboto', sans-serif;
}

.d-data-1 {
  width: 45%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.d-data-2 {
  color: white;

  h2 {
    font-weight: 500;
    font-size: 1.1rem;
  }

  p {
    font-size: 0.8rem;
    font-weight: 700;
    letter-spacing: 2px;
  }
}

.logo {
  height: 12vh;
  width: 12vh;
}
