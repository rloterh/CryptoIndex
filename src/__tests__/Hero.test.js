import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import * as reactRedux from 'react-redux';
import renderer from 'react-test-renderer';
import Hero from '../components/Hero/Hero';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Hero', () => {
  const useSelectorMock = reactRedux.useSelector;
  const useDispatchMock = reactRedux.useDispatch;
  const mockStore = {
    currencyReducer: [
      {
        id: '1',
        name: 'BitCoin-Test',
        symbol: 'BTC-TEST',
        priceUsd: '100',
        priceBtc: '1',
        marketCap: '32000',
        totalSupply: '333',
        rank: 1,
      },
    ],
    globalReducer: [
      {
        btcDominance: '43',
        ethDominance: '19',
      },
    ],
  };
  beforeEach(() => {
    useDispatchMock.mockImplementation(() => () => {});
    useSelectorMock.mockImplementation((selector) => selector(mockStore));
  });
  afterEach(() => {
    useDispatchMock.mockClear();
    useSelectorMock.mockClear();
  });

  test('Create and show list of Currencys', () => {
    render(<Hero />);
    expect(screen.getByTestId('global-container')).toBeInTheDocument();
  });
  test('Create and display GlobalCrypto Data component', () => {
    const globalData = useSelectorMock(
      (mockStore) => mockStore.globalReducer[0],
    );
    render(
      <Hero
        key={globalData.btcDominance}
        btcDominance={globalData.btcDominance}
        ethDominance={globalData.ethDominance}
      />,
    );
    expect(screen.getByTestId('global-container')).toBeInTheDocument();
  });
  test('Hero should render correctly', () => {
    const tree = renderer
      .create(
        <Hero />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
