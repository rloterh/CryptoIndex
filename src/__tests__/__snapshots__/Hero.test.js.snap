// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Hero should render correctly 1`] = `
<div
  className="hero-container d-flex"
  data-testid="global-container"
>
  <div
    className="d-data-1"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </div>
  <div
    className="d-data-2"
  >
    <h2>
      ::Currency Dominance::
    </h2>
    <p>
      BTC: 
      %
    </p>
    <p>
      ETH: 
      %
    </p>
  </div>
</div>
`;
